functions/src/export-csv.ts

```diff
@@ -167,3 +171,31 @@ function getFileName(jobName: string) {
   const fileBase = jobName.toLowerCase().replace(/[^a-z0-9]/gi, '-');
   return `${fileBase}.csv`;
 }
+
+function getAllPropertiesName(
+  lois: FirebaseFirestore.QuerySnapshot<FirebaseFirestore.DocumentData>): string[] {
+  let properties: string[] = [];
+  lois.forEach((loi) => {
```

Comment:
I think a functional expression might be more readable here. Also, we should prefer to return an immutable objects and data structures (ie immutable-js Set) where possible. So we'd get something like:

```ts
return Set(lois.map { loi => Object.keys(loi.get('properties')) || [] }.flat());
```